pkgs/tools/filesystems/btrfs-progs/default.nix

```diff
@@ -1,29 +1,57 @@
-{ lib, stdenv, fetchurl, pkg-config, attr, acl, zlib, libuuid, e2fsprogs, lzo
-, asciidoc, xmlto, docbook_xml_dtd_45, docbook_xsl, libxslt, zstd, python3
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoconf
+, automake
+, pkg-config
+, attr
+, acl
+, zlib
+, libuuid
+, e2fsprogs
+, lzo
+, asciidoc
+, xmlto
+, docbook_xml_dtd_45
+, docbook_xsl
+, libxslt
+, zstd
+, python3
 }:
 
 stdenv.mkDerivation rec {
   pname = "btrfs-progs";
   version = "5.12.1";
 
-  src = fetchurl {
-    url = "mirror://kernel/linux/kernel/people/kdave/btrfs-progs/btrfs-progs-v${version}.tar.xz";
-    sha256 = "sha256-lQhG/qRU+0scOfD6RUmDZEVy35HfXAYEezNb8tVHN1k=";
+  src = fetchFromGitHub {
+    owner = "kdave";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-rC9X4XzmT6PUDCajLkhuG85nRjJTqQ4mvevF4HWgHNE=";
   };
 
   nativeBuildInputs = [
-    pkg-config asciidoc xmlto docbook_xml_dtd_45 docbook_xsl libxslt
-    python3 python3.pkgs.setuptools
+    autoconf
+    automake
+    pkg-config
+    asciidoc
+    xmlto
+    docbook_xml_dtd_45
+    docbook_xsl
+    libxslt
+    python3
+    python3.pkgs.setuptools
   ];
 
   buildInputs = [ attr acl zlib libuuid e2fsprogs lzo zstd python3 ];
 
   # for python cross-compiling
   _PYTHON_HOST_PLATFORM = stdenv.hostPlatform.config;
 
-  # gcc bug with -O1 on ARM with gcc 4.8
-  # This should be fine on all platforms so apply universally
-  postPatch = "sed -i s/-O1/-O2/ configure";
+  preConfigure = ''
+    patchShebangs autogen.sh
+    sh autogen.sh
+  '';
```

Comment:
Can we simplify this with autoreconfHook?

Reply:
Unfortunately not.

Reply:
How about
```suggestion
  preAutoreconf = ''
    patchShebangs autogen.sh
  '';
```
?

Reply:
auroreconfHook doesn't work because `autoconf --force` returns this:

```
configure.ac:7: error: possibly undefined macro: AC_DEFINE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.ac:246: error: possibly undefined macro: AC_MSG_WARN
```

Running autogen.sh is the only way to get the source to configure AFAICT.